apdbPlots.py

```diff
@@ -0,0 +1,599 @@
+    ax1.spines['left'].set_visible(False)
+    ax2.spines['top'].set_visible(False)
+    ax2.spines['right'].set_visible(False)
+    ax1.invert_xaxis()
+    ax2.invert_xaxis()
+
+    ax1.yaxis.tick_right()
+    ax1.yaxis.set_label_position('right')
+
+    plt.title(title)
+    plt.subplots_adjust(wspace=0.1, hspace=0)
+
+
+def plotSourceLocationsOnObject(objId, objTable, sourceTable, repo, goodSrc=None, plotIdx=1):
+    """Plot a difference imaging template used for a DIA Object in a postage
+    stamp cutout, along with the locations of the DIA Object and all
+    constituent DIA Sources.
+
+    Parameters
+    ----------
+    objId : `int`
+        DIA Object ID.
+    objTable : `pandas.core.frame.DataFrame`
+        DIA Object table loaded from the APDB.
+    sourceTable : `pandas.core.frame.DataFrame`
+        Pandas dataframe with DIA Sources from an APDB, plus unpacked flag values.
+        This is an output called "sourceTableFlags" from the makeSrcTableFlags function.
+    repo : `str`
+        Repository corresponding to the output of an ap_pipe run.
+    goodSrc : `pandas.core.frame.DataFrame` or `None`
+        Filtered subset of sourceTable. Sources in this table will be plotted
+        in green, while those not in this table will be plotted in red.
+        If `None`, all sources will be plotted in blue.
+    plotIdx : `int`
+        Frame number to put the resulting plot in, default 1.
+
+    Notes
+    -----
+    The sourceTable is assumed to have flag fields added to it already!
+    """
+    butler = dafPersist.Butler(repo)
+    afwDisplay.setDefaultBackend('matplotlib')
+
+    objRa = objTable.loc[objTable['diaObjectId'] == objId, 'ra'].values[0]
+    objDec = objTable.loc[objTable['diaObjectId'] == objId, 'decl'].values[0]
+    srcIds = list(sourceTable.loc[sourceTable['diaObjectId'] == objId, 'diaSourceId'].values)
+    srcRas = sourceTable.loc[sourceTable['diaObjectId'] == objId, 'ra'].values
+    srcDecs = sourceTable.loc[sourceTable['diaObjectId'] == objId, 'decl'].values
+    srcCcdVisitIds = sourceTable.loc[sourceTable['diaObjectId'] == objId, 'ccdVisitId'].values
+    dataIds = [{'visit': int(str(dataId)[0:6]), 'ccdnum': int(str(dataId)[6:])} for dataId in srcCcdVisitIds]
+
+    objLoc = lsst.geom.SpherePoint(objRa, objDec, lsst.geom.degrees)
+    calexpFirst = butler.get('calexp', dataId=dataIds[0])
+    templateCutout = plc.getTemplateCutout(calexpFirst, repo, objLoc)
+    wcs = templateCutout.getWcs()
+
+    disp = afwDisplay.Display(plotIdx, reopenPlot=True)
+    disp.setMaskTransparency(100)
+    disp.scale('asinh', 'zscale', Q=4)
+    disp.mtv(templateCutout, title=objId)
+
+    with disp.Buffering():  # obviously
+        coordObj = wcs.skyToPixel(objLoc)
+        disp.dot('+', *coordObj, ctype='C0', size=5)
+        for dataId, srcId, ra, dec in zip(dataIds, srcIds, srcRas, srcDecs):
+            calexp = butler.get('calexp', dataId=dataId)
+            psf = calexp.getPsf()
```

Comment:
Is this information stored in a catalogue somewhere to prevent having to do all the butler calls? Should the PSF be position dependent on the CCD?

Reply:
Hmm. The PSF size definitely varies from visit to visit, and less so from CCD to CCD, as you say. If I really needed this to be fast, I could maybe compute a typical PSF size for a few CCDs in each visit and make sure they're ~similar and then adopt that. But this plot really doesn't need to be fast. I don't think information about the PSF is saved in any summary catalog (definitely not the APDB where most of the info I'm plotting is coming from).